torchaudio/csrc/lfilter.cpp

```diff
@@ -80,170 +80,161 @@ void lfilter_core_generic_loop(
   }
 }
 
-std::tuple<at::Tensor, at::Tensor> lfilter_core(
-    const torch::Tensor& waveform,
-    const torch::Tensor& a_coeffs,
-    const torch::Tensor& b_coeffs) {
-  TORCH_CHECK(waveform.device() == a_coeffs.device());
-  TORCH_CHECK(b_coeffs.device() == a_coeffs.device());
-  TORCH_CHECK(a_coeffs.size(0) == b_coeffs.size(0));
+class DifferentiableIIR : public torch::autograd::Function<DifferentiableIIR> {
+ public:
+  static torch::Tensor forward(
+      torch::autograd::AutogradContext* ctx,
+      const torch::Tensor& waveform,
+      const torch::Tensor& a_coeffs_normalized) {
+    at::AutoNonVariableTypeMode g;
```

Comment:
You definitely don't want that guard here!
This already runs in no_grad mode as it should.

Looks like you have a lot of them everywhere though...

Reply:
Thanks for explanation. I think we can just remove it in this merge.